src/psyclone/psyir/nodes/dynamic_omp_task_directive.py

```diff
@@ -1463,26 +1430,28 @@ def _evaluate_assignment(
             if isinstance(ref.parent, ArrayMixin):
                 continue
             for index, parent_var in enumerate(self._parent_loop_vars):
-                if ref.symbol == parent_var:
-                    if lhs.symbol in self._proxy_loop_vars:
-                        if (
-                            rhs
-                            not in self._proxy_loop_vars[lhs.symbol][
-                                "parent_node"
-                            ]
-                        ):
-                            self._proxy_loop_vars[lhs.symbol][
-                                "parent_node"
-                            ].append(rhs.copy())
-                    else:
-                        subdict = {}
-                        subdict["parent_var"] = parent_var
-                        subdict["parent_node"] = [rhs.copy()]
-                        subdict["loop"] = node
-                        subdict["parent_loop"] = self._parent_loops[index]
+                if ref.symbol != parent_var:
+                    continue
+                if lhs.symbol in self._proxy_loop_vars:
+                    if (
+                        rhs
+                        not in self._proxy_loop_vars[lhs.symbol].parent_node
+                    ):
+                        self._proxy_loop_vars[lhs.symbol].parent_node.append(
+                                rhs.copy()
+                        )
+                else:
+                    ProxyVars = namedtuple(
```

Comment:
Thanks for doing this. I think you can hoist the definition of the named tuple up out of the loop.

Reply:
Moved the definition of the named tuple to the class - and I'll make it private too.